src/Files.App/Utils/Git/GitHelpers.cs

```diff
@@ -33,6 +33,8 @@ internal static class GitHelpers
 
 		private static readonly IApplicationService _applicationService = Ioc.Default.GetRequiredService<IApplicationService>();
 
+		private static readonly ThreadWithMessageQueue _owningThread = new();
```

Comment:
I don't think keep the thread static is a good approach. Could we make it so that we only keep the thread when we do Git operations?

Reply:
I changed the code, the thread is static but when git operations finish, it will be disposed. This way it won't create multiple threads if more actions are performed at the same time (like when you navigate to a repo {fetch from origin, get branches and get head name})